### PR TITLE
Health check

### PR DESCRIPTION
**Pull request by lenak25**

### CODE REVIEW

File: src/worker/EnigmaNode.js [hunk 1]
```diff
@@ -491,15 +491,14 @@ class EnigmaNode extends EventEmitter {
...
       if (connectionErr) {
         this._logger.error(`Failed connecting to a bootstrap node= ${connectionErr}`);
-        return false;
       } else {
         this.notify(peerInfo);
-        return true;
       }
+      callback(connectionErr);
     });
   }
```
Comment (assafmo) [inline src/worker/EnigmaNode.js]: Please use promisify:
```javascript
const { promisify } = require("util");
// ...
connectToBootstrap(peerInfo) {
  const dial = promisify(dial).bind(this);
  try {
    const connection = await dial(peerInfo); // Notice that connection is unused
  } catch (connectionErr) {
    this._logger.error(`Failed connecting to a bootstrap node= ${connectionErr}`);
    return false;
  }

  this.notify(peerInfo);
  return true;
}
```

Comment (assafmo) [reply src/worker/EnigmaNode.js]: Also Notice that `dial` can be implemented with `promisify` on `dialProtocol` and `node.dialProtocol`.  
This way you won't have to do `const dial = promisify(dial).bind(this);`.

File: src/worker/controller/NodeController.js [hunk 2]
```diff
@@ -709,5 +803,67 @@ class NodeController {
...
+      // TODO: consider adding a periodic once there
+      /*state: {
+        status: false,
+        missing: null
+      }*/
+    };
+
+    // core
+    try {
+      let regParams = await this.asyncGetRegistrationParams();
```
Comment (assafmo) [inline src/worker/controller/NodeController.js]: let -> const

File: src/worker/controller/NodeController.js [hunk 3]
```diff
@@ -709,5 +803,67 @@ class NodeController {
...
+      healthCheckResult.core.registrationParams.signKey = regParams.result.signingKey;
+      healthCheckResult.core.status = healthCheckResult.core.registrationParams.signKey != null;
+    } catch (e) {
+      healthCheckResult.core.status = false;
+    }
+
+    // ethereum
+    if (this.hasEthereum()) {
+      try {
+        let eth = await this.ethereum().healthCheck();
```
Comment (assafmo) [inline src/worker/controller/NodeController.js]: let -> const

File: src/worker/controller/actions/HealthCheckAction.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    };
+
+    this._controller.execCmd(C.REGISTRATION_PARAMS, {
+      onResponse: async (err, regParams) => {
+        if (!err) {
+          healthCheckResult.core.registrationParams.signKey = regParams.result.signingKey;
+          healthCheckResult.core.status = healthCheckResult.core.registrationParams.signKey != null;
+          if (this._controller.hasEthereum()) {
+            try {
+              let eth = await this._controller.ethereum().healthCheck();
```
Comment (assafmo) [inline src/worker/controller/actions/HealthCheckAction.js:42]: const

File: src/worker/controller/actions/connectivity/BootstrapDiscoveredAction.js [hunk 5]
```diff
@@ -11,8 +11,13 @@ class BootstrapDiscoveredAction {
...
     }
-    const success = this._controller.engNode().connectToBootstrap(otherPeer);
-    this._controller.logger().debug(`connection to bootstrap succeeded=${success}`);
+    this._controller.engNode().connectToBootstrap(otherPeer, err => {
+      let success = false;
+      if (!err) {
+        success = true;
+      }
+      this._controller.logger().debug(`connection to bootstrap succeeded=${success}`);
+    });
```
Comment (assafmo) [inline src/worker/controller/actions/connectivity/BootstrapDiscoveredAction.js]: Please use `promisify` with `try/catch`.

**Review by assafmo — CHANGES_REQUESTED** [review]

**Comment by assafmo** [comment]
@lenak25  You can ignore the "let -> const" comments, but please use promisify, it will greatly improve code readability. :pray:

**Comment by lenak25** [comment]
@assafmo , code review comments were integrated.. please review again :)

**Review by assafmo — DISMISSED** [review]

**Comment by assafmo** [comment]
@lenak25 Re: https://github.com/enigmampc/enigma-p2p/issues/262. Please add basic documentation (`curl` examples should suffice for now).

**Comment by assafmo** [comment]
Also "Merging is blocked" due to "The base branch requires all commits to be signed." :man_shrugging:  

Try squashing this entire PR or resign every commit since branching out:  
https://superuser.com/a/1123928/725752
```bash
git rebase --exec 'git commit --amend --no-edit -n -S' -i develop
```

**Review by assafmo — APPROVED** [review]

**Pull request merged**